src/core/markovProcessGraph.js

```diff
@@ -0,0 +1,605 @@
+          owner: scoringAddress,
+          epochStart: boundary,
+        });
+        addNode({
+          address: thisEpoch,
+          description: `Epoch starting ${boundary} ms past epoch`,
+          mint: 0,
+        });
+        addEdge({
+          address: EdgeAddress.append(
+            EPOCH_PAYOUT,
+            String(boundary),
+            ...NodeAddress.toParts(scoringAddress)
+          ),
+          reversed: false,
+          src: thisEpoch,
+          dst: scoringAddress,
+          transitionProbability: fibration.beta,
+        });
+        if (lastBoundary != null) {
+          const lastEpoch = epochNodeAddressToRaw({
+            type: "EPOCH_NODE",
+            owner: scoringAddress,
+            epochStart: lastBoundary,
+          });
+          const webAddress = EdgeAddress.append(
+            EPOCH_WEBBING,
+            String(boundary),
+            ...NodeAddress.toParts(scoringAddress)
+          );
+          addEdge({
+            address: webAddress,
+            reversed: false,
+            src: lastEpoch,
+            dst: thisEpoch,
+            transitionProbability: fibration.gammaForward,
+          });
+          addEdge({
+            address: webAddress,
+            reversed: true,
+            src: thisEpoch,
+            dst: lastEpoch,
+            transitionProbability: fibration.gammaBackward,
+          });
+        }
+        lastBoundary = boundary;
+      }
+    }
+
+    // Add minting edges, from the seed to positive-weight graph nodes
+    {
+      let totalNodeWeight = 0.0;
+      const positiveNodeWeights: Map<NodeAddressT, number> = new Map();
+      for (const {address, mint} of _nodes.values()) {
+        if (mint > 0) {
+          totalNodeWeight += mint;
+          positiveNodeWeights.set(address, mint);
+        }
+      }
+      if (!(totalNodeWeight > 0)) {
+        throw new Error("No outflow from seed; add cred-minting nodes");
+      }
+      for (const [address, weight] of positiveNodeWeights) {
+        addEdge({
+          address: EdgeAddress.append(
+            SEED_MINT,
+            ...NodeAddress.toParts(address)
+          ),
+          reversed: false,
+          src: SEED_ADDRESS,
+          dst: address,
+          transitionProbability: weight / totalNodeWeight,
+        });
+      }
+    }
+
+    /**
+     * Find an epoch node, or just the original node if it's not a
+     * scoring address.
+     */
+    const rewriteEpochNode = (
+      address: NodeAddressT,
+      edgeTimestampMs: TimestampMs
+    ): NodeAddressT => {
+      if (!_scoringAddresses.has(address)) {
+        return address;
+      }
+      const epochEndIndex = sortedIndex(timeBoundaries, edgeTimestampMs);
+      const epochStartIndex = epochEndIndex - 1;
+      const epochTimestampMs = timeBoundaries[epochStartIndex];
+      return epochNodeAddressToRaw({
+        type: "EPOCH_NODE",
+        owner: address,
+        epochStart: epochTimestampMs,
+      });
+    };
```

Comment:
Would be nice if we can get unit testing of this function in particular. Feels like if it were off-by-one our current test plan wouldn't catch it.